### PR TITLE
feat: add gem fiat ramp integration

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

reopened from #1062

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [ ] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/847

## Risk

`selectPortfolioCryptoHumanBalanceBySymbol` might cause collisions with assets that have the same symbol, but are actually different ones on the same chain, or even on a totally different chain.

## Testing

TODO

## Screenshots (if applicable)

TODO

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -17,6 +17,12 @@ jobs:
       REACT_APP_ETHEREUM_NODE_URL: https://mainnet.infura.io/v3/d734c7eebcdf400185d7eb67322a7e57
       REACT_APP_FEATURE_COSMOS_INVESTOR: true
       REACT_APP_FEATURE_PLUGIN_COSMOS: true
+      REACT_APP_GEM_COINIFY_SUPPORTED_COINS: https://api.gem.co/institutions/coinify/supported_currencies
+      REACT_APP_GEM_WYRE_SUPPORTED_COINS: https://api.gem.co/institutions/wyre/supported_currencies
+      REACT_APP_GEM_ASSET_LOGO: https://gem-widgets-assets.s3-us-west-2.amazonaws.com/currencies/crypto/
+      REACT_APP_GEM_ENV: production
+      REACT_APP_GEM_API_KEY: 17baca45f492df6b8974d4f832a659c92644acafea3284cfb0d52550dd3bde1a
+      REACT_APP_GEM_URL: https://onramp.gem.co
```
Comment (gomesalexandre) [inline .github/workflows/pr.yml:25]: How do we feel about these? IMO, since this is a public, external endpoint, we do not need to "hide" it into ENV

**Comment by gomesalexandre** [comment]
Still working on refactoring this (moving stuff away in utils/reducer) as we speak, but in the current state, it is ready for general logic review

File: .github/workflows/pr.yml (hunk 1 above)
Comment (0xdef1cafe) [reply .github/workflows/pr.yml:25]: this is just an id not a secret, nothing is hidden

File: src/components/Layout/Header/Header.tsx [hunk 2]
```diff
@@ -43,6 +46,8 @@ export const Header = ({ route }: { route: Route }) => {
     return () => document.removeEventListener('keydown', handleKeyPress)
   }, [handleKeyPress])
 
+  const gemRampFlag = useSelector(state => selectFeatureFlag(state as any, 'GemRamp'))
```
Comment (0xdef1cafe) [inline src/components/Layout/Header/Header.tsx:49]:
```suggestion
  const gemRampFlag = useAppSelector(state => selectFeatureFlag(state, 'GemRamp'))
```

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

File: src/components/Layout/Header/Header.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/components/Layout/Header/Header.tsx:49]: I initially went with this, but we get this bad boy error:
```
portfolioSlice.ts:170 Uncaught ReferenceError: Cannot access 'selectAccountIdByAddress' before initialization
```

I grepped `selectAccountIdByAddress` and couldn't find what caused it. `useAppSelector` should simply be a RTK-typed alias for `useSelector` so this one is weird. Can you reproduce it?

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by mrnerdhair** [comment]
Followed up in PR #1194; skipped making issue for these since they're so trivial.

**Pull request merged**